src/pages/admin/notices/edit/[id].vue

```diff
@@ -0,0 +1,150 @@
+        <div class="mb-3 flex">
+          <div>
+            <AppInput v-model="data.expirationDate" type="date" />
+          </div>
+          <div>
+            <AppInput v-model="data.expirationTime" type="time" />
+          </div>
+        </div>
+
+        <div class="mb-3">
+          <AppInput
+            v-model="data.text"
+            :label="t('addNotice.form.text')"
+            required
+          />
+        </div>
+
+        <div class="mb-3">
+          <AppInput
+            v-model="data.buttonText"
+            :label="t('addNotice.form.buttonText')"
+          />
+        </div>
+
+        <div class="mb-3">
+          <AppInput v-model="data.url" :label="t('addNotice.form.url')" />
+        </div>
+      </AppForm>
+    </div>
+  </template>
+</template>
+
+<script lang="ts" setup>
+import { format } from 'date-fns';
+import { ref, onBeforeMount, reactive } from 'vue';
+import { useI18n } from 'vue-i18n';
+import { useRouter } from 'vue-router';
+
+import AppForm from '../../../../components/forms/AppForm.vue';
+import AppInput from '../../../../components/forms/AppInput.vue';
+import AppLabel from '../../../../components/forms/AppLabel.vue';
+import PageTitle from '../../../../components/PageTitle.vue';
+import {
+  CreateNoticeData,
+  GetNoticeData,
+} from '../../../../utils/api/api.interface';
+import {
+  updateNotice,
+  fetchNotice,
+  createNotice,
+} from '../../../../utils/api/notice';
+
+const { t } = useI18n();
+const router = useRouter();
+
+const props = defineProps<{ id: string }>();
+const notice = ref<GetNoticeData | undefined>();
+
+const data = reactive({
+  name: '',
+  startDate: '',
+  startTime: '',
+  expirationDate: '',
+  expirationTime: '',
+  text: '',
+  buttonText: '',
+  url: '',
+});
+
+onBeforeMount(async () => {
+  notice.value = await fetchNotice(props.id as string);
+
+  data.name = notice.value.name;
+  data.startDate = notice.value.starts
+    ? format(notice.value.starts, 'yyyy-MM-dd')
+    : '';
+  data.startTime = notice.value.starts
+    ? format(notice.value.starts, 'HH:mm')
+    : '';
+  data.expirationDate = notice.value.expires
+    ? format(notice.value.expires, 'yyyy-MM-dd')
+    : '';
+  data.expirationTime = notice.value.expires
+    ? format(notice.value.expires, 'HH:mm')
+    : '';
+  data.text = notice.value.text;
+  data.buttonText = notice.value.buttonText ? notice.value.buttonText : '';
+  data.url = notice.value.url ? notice.value.url : '';
+});
+
+async function saveNotice() {
+  const noticeToSubmit: CreateNoticeData = {
+    name: data.name,
+    starts: new Date(data.startDate + 'T' + data.startTime),
+    expires: new Date(data.expirationDate + 'T' + data.expirationTime),
+    text: data.text,
+    buttonText: data.buttonText,
+    url: data.url,
+  };
+  if (notice.value) {
+    return await updateNotice(notice.value.id, noticeToSubmit);
+  } else {
+    return await createNotice(noticeToSubmit);
```

Comment:
Not sure I understand this as this page can only ever be loaded for editing a notice.

`saveNotice` can't be called unless notice is defined as the form won't show, although unfortunately the compiler doesn't know this. I would usually solve this with a quick return at the top of the method and a comment

```ts
async function saveNotice() {
  if (!notice.value) return; // Can't submit without notice being loaded
  ...
}
```